content/en/blog/_posts/2018-07-10-coredns-ga.md

```diff
@@ -0,0 +1,179 @@
+
+CoreDNS is a general-purpose, authoritative DNS server that provides a backwards-compatible, but extensible integration with Kubernetes. It resolves the issues seen with `kube-dns`, and offers a number of unique features that solve a wider variety of use cases.
+
+In this article, you will learn about the differences in the implementations of `kube-dns` and CoreDNS, and some of the helpful extensions offered by CoreDNS.
+
+## Implemenation differences
+
+In `kube-dns`, several containers are used within a single pod: `kubedns`, `dnsmasq`, and `sidecar`. The `kubedns`
+container watches the Kubernetes API and serves DNS records based on the [Kubernetes DNS specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md), while `dnsmasq` provides caching and stub domain support, and `sidecar` provides metrics and health checks.
+
+This setup leads to a few issues that have been seen over time. For one, since `dnsmasq` is written in C, it is not
+memory-safe. This means it is more prone to certain types of security vulnerabilities, and that has led to the need
+for a security-patch release of Kubernetes in the past. Additionally, because `dnsmasq` handles the stub domains,
+but `kubedns` handles the External Services, you cannot use a stub domain in an external service, which is very
+limiting to that functionality (see [dns#131](https://github.com/kubernetes/dns/issues/131)).
+
+All of these functions are done in a single container in CoreDNS, which is running a process written in Go. The
+different plugins that are enabled replicate (and enhance) the functionality found in `kube-dns`.
+
+## Configuring CoreDNS
+
+In `kube-dns`, you can [modify a ConfigMap](https://kubernetes.io/blog/2017/04/configuring-private-dns-zones-upstream-nameservers-kubernetes/) to change the behavior of your service discovery. This allows the addition of
+features such as serving stub domains, modifying upstream nameservers, and enabling federation.
+
+In CoreDNS, you similarly can modify the ConfigMap for the CoreDNS [Corefile](https://coredns.io/2017/07/23/corefile-explained/) to change how service discovery
+works. This Corefile configuration offers many more options than you will find in `kube-dns`, since it is the
+primary configuration file that CoreDNS uses for configuration of all of its features, even those that are not
+Kubernetes related.
+
+When upgrading from `kube-dns` to CoreDNS using `kubeadm`, your existing ConfigMap will be used to generate the
+customized Corefile for you, including all of the configuration for stub domains, federation, and upstream nameservers. See [Using CoreDNS for Service Discovery](https://kubernetes.io/docs/tasks/administer-cluster/coredns/) for more details.
+
+
+## Bug fixes and enhancements
+
+There are several open issues with `kube-dns` that are resolved in CoreDNS, including:
```

Comment:
How are these solved? They just work I do I need to fiddle with the Configmap/Corefile. Can we point to CoreDNS issues/PRs ?